### PR TITLE
Fix fetcher tests

### PR DESCRIPTION
**Pull request by subhramit**
Fixes:
- `OpenAccessDoiTest`
- `TitleFetcherTest`
### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Review by Siedlerchr — DISMISSED** [review]

File: src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java [hunk 1]
```diff
@@ -28,7 +28,7 @@ void setUp() {
     @Test
     void findByDOI() throws IOException {
         entry.setField(StandardField.DOI, "10.1038/nature12373");
-        assertEquals(Optional.of(new URL("https://dash.harvard.edu/bitstream/1/12285462/1/Nanometer-Scale%20Thermometry.pdf")), finder.findFullText(entry));
+        assertEquals(Optional.of(new URL("https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4221854")), finder.findFullText(entry));
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java:31]: This should return a fulltext and should therefore end in pdf

Comment (subhramit) [reply src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java:31]: Check the expected value in https://github.com/JabRef/jabref/actions/runs/10771322038/job/29866598073.
The link does not lead to a pdf.

Comment (subhramit) [reply src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java:31]: So is there something wrong in `findFullText()`?

Comment (Siedlerchr) [reply src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java:31]: Yes, that's what I was trying to say. findFullText should point to a pdf file that can be downloaded.
So the OpenAccessDOI fetcher neeeds to be adjusted

Comment (subhramit) [reply src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java:31]: Reverting the adapted change for `OpenAccessDoiFetcher` then for this PR.

Comment (subhramit) [reply src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java]: > Edit // https://unpaywall.org/data-format Seems like the fetcher is correct, ist just doesn't return a pdf anymore for that specific article

Let me find another article and adapt then.

Comment (subhramit) [reply src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java]: Weird. Still the same behavior.

Comment (subhramit) [reply src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java]: Okay, this time, it does link to a pdf, but the link doesn't end with `.pdf`.

**Review by Siedlerchr — DISMISSED** [review]

File: src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java [hunk 2]
```diff
@@ -27,8 +27,8 @@ void setUp() {
 
     @Test
     void findByDOI() throws IOException {
-        entry.setField(StandardField.DOI, "10.1038/nature12373");
-        assertEquals(Optional.of(new URL("https://dash.harvard.edu/bitstream/1/12285462/1/Nanometer-Scale%20Thermometry.pdf")), finder.findFullText(entry));
+        entry.setField(StandardField.DOI, "10.1186/s12993-024-00248-9");
+        assertEquals(Optional.of(new URL("https://behavioralandbrainfunctions.biomedcentral.com/counter/pdf/10.1186/s12993-024-00248-9.pdf")), finder.findFullText(entry));
```
Comment (Siedlerchr) [inline src/test/java/org/jabref/logic/importer/fetcher/OpenAccessDoiTest.java:31]: So remove the .pdf extension here?

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**